test/src/textureTest.cpp

```diff
@@ -0,0 +1,145 @@
+
+#ifdef RGL_BUILD_ROS2_EXTENSION
+#include <rgl/api/extensions/ros2.h>
+#endif
+
+
+struct TextureTest : public RGLTestWithParam<std::tuple<int, int, float>>, public RGLPointTestHelper{};
+
+INSTANTIATE_TEST_SUITE_P(
+		Parametrized, TextureTest,
+		testing::Combine(
+				testing::Values(1, 100, 4000),
+				testing::Values(2, 200, 2000),
+				testing::Values(-4.2f, 0.0f, 21.37f)
+		));
+
+TEST_F(TextureTest, rgl_texture_invalid_argument)
+{
+	rgl_texture_t texture= nullptr;
+	float* textureRawData = generateStaticColorTexture(100, 100, 100.0f);
+
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_texture_create(nullptr, textureRawData, 100, 100), "texture != nullptr");
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_texture_create(&texture, nullptr, 100, 100), "texels != nullptr");
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_texture_create(&texture, textureRawData, -1, 100), "width > 0");
+	EXPECT_RGL_INVALID_ARGUMENT(rgl_texture_create(&texture, textureRawData, 100, 0), "height > 0");
+
+	delete[] textureRawData;
+}
+
+TEST_P(TextureTest, rgl_texture_reading)
+{
+	auto [width, height, value] = GetParam();
+
+	rgl_texture_t texture= nullptr;
+	rgl_entity_t entity = nullptr;
+	rgl_mesh_t mesh = makeCubeMesh();
+	auto* textureRawData = generateStaticColorTexture(width, height, value);
+
+	EXPECT_RGL_SUCCESS(rgl_texture_create(&texture, textureRawData, width, height));
+	EXPECT_RGL_SUCCESS(rgl_mesh_set_texture_coords(mesh, cubeUVs, 8));
+
+	EXPECT_RGL_SUCCESS(rgl_entity_create(&entity, nullptr, mesh));
+	EXPECT_RGL_SUCCESS(rgl_entity_set_intensity_texture(entity, texture));
+
+	// Create RGL graph pipeline.
+	rgl_node_t useRaysNode = nullptr, raytraceNode = nullptr, compactNode = nullptr, yieldNode = nullptr;
+
+	std::vector<rgl_mat3x4f> rays = makeLidar3dRays(360, 360, 0.36, 0.36);
+
+	std::vector<rgl_field_t> yieldFields = {
+			INTENSITY_F32
+	};
+
+	EXPECT_RGL_SUCCESS(rgl_node_rays_from_mat3x4f(&useRaysNode, rays.data(), rays.size()));
+	EXPECT_RGL_SUCCESS(rgl_node_raytrace(&raytraceNode, nullptr, 1000));
+	EXPECT_RGL_SUCCESS(rgl_node_points_compact(&compactNode));
+	EXPECT_RGL_SUCCESS(rgl_node_points_yield(&yieldNode, yieldFields.data(), yieldFields.size()));
+
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(useRaysNode, raytraceNode));
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(raytraceNode, compactNode));
+	EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(compactNode, yieldNode));
+
+	EXPECT_RGL_SUCCESS(rgl_graph_run(raytraceNode));
+
+	std::vector<::Field<INTENSITY_F32>::type> outIntensity;
+
+	int32_t outCount, outSizeOf;
+	EXPECT_RGL_SUCCESS(rgl_graph_get_result_size(yieldNode, INTENSITY_F32, &outCount, &outSizeOf));
+	EXPECT_EQ(outSizeOf, getFieldSize(INTENSITY_F32));
+
+	outIntensity.resize(outCount);
+
+	EXPECT_RGL_SUCCESS(rgl_graph_get_result_data(yieldNode, INTENSITY_F32, outIntensity.data()));
+
+	for (int i = 0; i < outCount; ++i)
+	{
+		EXPECT_NEAR(value, outIntensity.at(i), EPSILON_F);
+	}
+
+	delete[] textureRawData;
+}
+
+
+TEST_P(TextureTest, rgl_texture_use_case)
```

Comment:
Please add a comment before the test to explain what is going on here, what to expect, and how to check if the test passes.